genvm/core/context.go

```diff
@@ -164,9 +225,38 @@ func (c *Context) load(address types.Address) (*Account, error) {
 }
 
 func (c *Context) change(account *Account) {
+	if account.Address == c.Principal() {
+		return
+	}
 	_, exist := c.changed[account.Address]
 	if !exist {
 		c.touched = append(c.touched, account.Address)
 	}
 	c.changed[account.Address] = account
 }
+
+// RemoteContext ...
+type RemoteContext struct {
+	*Context
+	remote   *Account
+	handler  Handler
+	template Template
+}
+
+// Template ...
+func (r *RemoteContext) Template() Template {
+	return r.template
+}
+
+// Handler ...
+func (r *RemoteContext) Handler() Handler {
+	return r.handler
+}
+
+// Transfer ...
+func (r *RemoteContext) Transfer(to Address, amount uint64) error {
+	if err := r.transfer(r.remote, to, amount, r.remote.Balance); err != nil {
```

Comment:
should maxSpend here be `amount` instead?

Reply:
doesn't matter really, can be amount